lib/device/library/windows/eject.rb

```diff
@@ -0,0 +1,166 @@
+        szVolumeFormat = "\\\\.\\%s:"
+        szRootFormat = "%s:\\"
+
+        szRootName = szRootFormat % cDriveLetter
+
+        uDriveType = WinAPI.GetDriveType(szRootName)
+        case uDriveType
+        when DRIVE_REMOVABLE
+          dwAccessFlags = GENERIC_READ | GENERIC_WRITE
+        when DRIVE_CDROM
+          dwAccessFlags = GENERIC_READ
+        else
+          return INVALID_HANDLE_VALUE
+        end
+
+        szVolumeName = szVolumeFormat % cDriveLetter
+
+        hVolume = WinAPI.CreateFile(
+          szVolumeName,
+          dwAccessFlags,
+          FILE_SHARE_READ | FILE_SHARE_WRITE,
+          NULL,
+          OPEN_EXISTING,
+          0,
+          NULL
+        )
+        hVolume
+      end
+
+      def lockVolume(hVolume)
+        dwSleepAmount = LOCK_TIMEOUT / LOCK_RETRIES
+        dwBytesReturned = "    ".dup
+
+        # Do this in a loop until a timeout period has expired
+        LOCK_RETRIES.times do
+          res = WinAPI.DeviceIoControl(
+            hVolume,
+            FSCTL_LOCK_VOLUME,
+            NULL, 0,
+            NULL, 0,
+            dwBytesReturned,
+            NULL
+          )
+          return true if res != 0
+          sleep(dwSleepAmount)
+        end
+        raise Device::CantEject, "端末が使用中なため取り外せませんでした"
+      end
+
+      def dismountVolume(hVolume)
+        dwBytesReturned = "    ".dup
+        WinAPI.DeviceIoControl(
+          hVolume,
+          FSCTL_DISMOUNT_VOLUME,
+          NULL, 0,
+          NULL, 0,
+          dwBytesReturned,
+          NULL
+        ) != 0
+      end
+
+      def preventRemovalOfVolume(hVolume, fPreventRemoval)
+        dwBytesReturned = "    ".dup
+        WinAPI.DeviceIoControl(
+          hVolume,
+          IOCTL_STORAGE_MEDIA_REMOVAL,
+          [fPreventRemoval.to_s].pack("b"), 1,
+          NULL, 0,
+          dwBytesReturned,
+          NULL
+        ) != 0
+      end
+
+      def autoEjectVolume(hVolume)
+        dwBytesReturned = "    ".dup
+        WinAPI.DeviceIoControl(
+          hVolume,
+          IOCTL_STORAGE_EJECT_MEDIA,
+          NULL, 0,
+          NULL, 0,
+          dwBytesReturned,
+          NULL
+        ) != 0
+      end
+
+      def ejectVolume(cDriveLetter)
+        fRemoveSafely = false
+        fAutoEject = false
+        hVolume = INVALID_HANDLE_VALUE
+
+        cDriveLetter = cDriveLetter[0]
+        unless cDriveLetter
+          raise Device::CantEject, "need a drive letter"
+        end
+
+        # Open the volume.
+        hVolume = openVolume(cDriveLetter)
+        if hVolume == INVALID_HANDLE_VALUE
+          raise Device::CantEject, "Cannot eject. Drive type is incorrect."
+        end
+
+        # Lock and dismount the volume.
+        if lockVolume(hVolume) && dismountVolume(hVolume)
+          fRemoveSafely = true
+
+          # Set prevent removal to false and eject the volume.
+          if preventRemovalOfVolume(hVolume, WIN_FALSE) && autoEjectVolume(hVolume)
+            fAutoEject = true
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'fAutoEject' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)